From b9e1f11a2da60b5bb7e79acbc4c48fe8b342a032 Mon Sep 17 00:00:00 2001
From: Soren Hansen <soren@linux2go.dk>
Date: Thu, 12 Aug 2010 15:06:34 +0200
Subject: [PATCH] Make umlConnectTapDevice ask brAddTap for a persistent tap device.

This patch does two things:

 * It makes umlConnectTapDevice ask brAddTap for a persistent tap by
   passing it a NULL tapfd argument.
 * Stops umlConnectTapDevice from immediately dismantling the bridge
   it just set up.

Signed-off-by: Soren Hansen <soren@linux2go.dk>
---
 src/uml/uml_conf.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

Index: libvirt-0.8.3/src/uml/uml_conf.c
===================================================================
--- libvirt-0.8.3.orig/src/uml/uml_conf.c	2010-08-16 13:30:12.636132000 +0200
+++ libvirt-0.8.3/src/uml/uml_conf.c	2010-08-16 13:31:17.076132001 +0200
@@ -112,7 +112,6 @@
                     const char *bridge)
 {
     brControl *brctl = NULL;
-    int tapfd = -1;
     int template_ifname = 0;
     int err;
     unsigned char tapmac[VIR_MAC_BUFLEN];
@@ -140,7 +139,7 @@
                         &net->ifname,
                         tapmac,
                         0,
-                        &tapfd))) {
+                        NULL))) {
         if (errno == ENOTSUP) {
             /* In this particular case, give a better diagnostic. */
             umlReportError(VIR_ERR_INTERNAL_ERROR,
@@ -159,7 +158,6 @@
             VIR_FREE(net->ifname);
         goto error;
     }
-    close(tapfd);
 
     brShutdown(brctl);
 
